src/utils/vcs.rs

```diff
@@ -1719,10 +1711,7 @@ mod tests {
         })
         .to_string();
 
-        assert_eq!(extract_pr_base_sha_from_event(&push_json).unwrap(), None);
-
-        // Test with malformed JSON
-        assert!(extract_pr_base_sha_from_event("invalid json {").is_err());
```

Comment:
We should still test that the function handles invalid JSON correctly by panicking. You can do this by splitting this test into its own test function, and by [marking the separate test function with `#[should_panic]`](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic).